samples/bluetooth/df_per_adv_beacon/src/main.c

```diff
@@ -0,0 +1,208 @@
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <sys/printk.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/direction.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+/* Length of CTE in unit of 8[us] */
+#define CTE_LEN (0x14U)
+
+static void adv_sent_cb(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_sent_info *info);
+
+static struct bt_le_ext_adv_cb adv_callbacks = {
+	.sent = adv_sent_cb,
+};
+
+static struct bt_le_ext_adv *adv_set;
+
+static struct bt_le_adv_param param =
+		BT_LE_ADV_PARAM_INIT(BT_LE_ADV_OPT_EXT_ADV |
+				     BT_LE_ADV_OPT_USE_NAME,
+				     BT_GAP_ADV_FAST_INT_MIN_2,
+				     BT_GAP_ADV_FAST_INT_MAX_2,
+				     NULL);
+
+static struct bt_le_ext_adv_start_param ext_adv_start_param = {
+	.timeout = 0,
+	.num_events = 0,
+};
+
+static struct bt_le_per_adv_param per_adv_param = {
+	.interval_min = BT_GAP_ADV_SLOW_INT_MIN,
+	.interval_max = BT_GAP_ADV_SLOW_INT_MAX,
+	.options = BT_LE_ADV_OPT_USE_TX_POWER,
+};
+
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX)
+static uint8_t ant_patterns[] = {0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA };
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_TX */
+
+struct bt_df_adv_cte_tx_param cte_params = {
+	.cte_len = CTE_LEN,
+	.cte_count = 1,
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ANT_SWITCH_TX)
+	.cte_type = BT_HCI_LE_AOD_CTE_2US,
+	.num_ant_ids = ARRAY_SIZE(ant_patterns),
+	.ant_ids = ant_patterns
+#else
+	.cte_type = BT_HCI_LE_AOA_CTE,
+	.num_ant_ids = 0,
+	.ant_ids = NULL
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_TX */
+	};
+
+static void adv_sent_cb(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_sent_info *info)
+{
+	printk("Advertiser[%d] %p sent %d\n", bt_le_ext_adv_get_index(adv),
+	       adv, info->num_sent);
+}
+
+static struct bt_le_ext_adv *advertising_set_create(void)
+{
+	struct bt_le_ext_adv *adv;
+	int err;
+
+	printk("Advertising set create...");
+	err = bt_le_ext_adv_create(&param, &adv_callbacks, &adv);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return NULL;
+	}
+	printk("success\n");
+	return adv;
+}
+
+static int periodic_advertising_params_set(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	printk("Periodic advertising params set...");
+	err = bt_le_per_adv_set_param(adv, &per_adv_param);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	return 0;
+}
+
+static int periodic_advertising_enable(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	printk("Periodic advertising enable...");
+	err = bt_le_per_adv_start(adv);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	printk("Extended advertising enable...");
+	err = bt_le_ext_adv_start(adv, &ext_adv_start_param);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return err;
+	}
+	printk("success\n");
+
+	char addr_s[BT_ADDR_LE_STR_LEN];
+	bt_addr_le_t addr = {0};
+	size_t count = 1;
+
+	bt_id_get(&addr, &count);
```

Comment:
This would be incorrect if BT_PRIVACY was enabled.
Better to use `bt_le_ext_adv_oob_get_local` to retrieve the advertiser address.

Reply:
Thank you for the hint. Changed to use bt_le_ext_adv_oob_get_local.